src/keboola_mcp_server/tools/workspace.py

```diff
@@ -420,3 +420,24 @@ async def get_quoted_name(self, name: str) -> str:
     async def get_sql_dialect(self) -> str:
         workspace = await self._get_workspace()
         return workspace.get_sql_dialect()
+
+
+class ProjectManager:
+    STATE_KEY = 'project_manager'
+
+    @classmethod
+    def from_state(cls, state: Mapping[str, Any]) -> 'ProjectManager':
+        instance = state[cls.STATE_KEY]
+        assert isinstance(instance, ProjectManager), f'Expected ProjectManager, got: {instance}'
+        return instance
+
+    def __init__(self, client: KeboolaClient):
+        self.project = None
+        self.client = client
+
+    async def get_project_id(self) -> str | int:
+        if self.project is None:
+            # This should probably be handled for exceptions somehow - unclear how at this time
+            project_data = await self.client.storage_client.verify_token()
+            self.project = project_data.get('owner', {}).get('id')
```

Comment:
We shouldn't silently allow the project_id to be `None`. Also, `str(self.project)` would return string `'None'`.